mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala

```diff
@@ -397,10 +406,15 @@ object PowerIterationClustering extends Logging {
             s" difference delta = ${diffDelta} and norm = ${norm}")
         }
       }
+      curG.vertices.unpersist()
+      curG.edges.unpersist()
```

Comment:
nit: just `curG.unpersist`?

Reply:
Actually `curG.unpersist` doesn't seem unpersist the edges rdd. One `EdgeRDD` block is getting cached in every iteration. That is why I have explicitely uncached into both edges and vertices